presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java

```diff
@@ -287,36 +300,46 @@ public synchronized void scheduleRequest()
         requestsScheduled.incrementAndGet();
     }
 
-    private synchronized void initiateRequest()
+    private synchronized void initiateRequest(DataSize maxSize)
```

Comment:
This should be `synchronized` right? The function is invoked by an executor with a delay.